src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1473,46 +1473,52 @@ void SolverInterfaceImpl::computePartitions()
   }
 }
 
-void SolverInterfaceImpl::mapWrittenData()
+void SolverInterfaceImpl::computeMappings(utils::ptr_vector<MappingContext> contexts, std::string mappingType)
 {
-  PRECICE_TRACE();
   using namespace mapping;
   MappingConfiguration::Timing timing;
-  // Compute mappings
-  for (impl::MappingContext &context : _accessor->writeMappingContexts()) {
-    timing         = context.timing;
-    bool rightTime = timing == MappingConfiguration::ON_ADVANCE;
-    rightTime |= timing == MappingConfiguration::INITIAL;
+  for (impl::MappingContext &context : contexts) {
+    timing      = context.timing;
+    bool mapNow = timing == mapping::MappingConfiguration::ON_ADVANCE;
+    mapNow |= timing == mapping::MappingConfiguration::INITIAL;
     bool hasComputed = context.mapping->hasComputedMapping();
-    if (rightTime && not hasComputed) {
-      PRECICE_INFO("Compute write mapping from mesh \"{}\" to mesh \"{}\".",
-                   _accessor->meshContext(context.fromMeshID).mesh->getName(), _accessor->meshContext(context.toMeshID).mesh->getName());
+    if (mapNow && not hasComputed) {
+      PRECICE_INFO("Compute \"{}\" mapping from mesh \"{}\" to mesh \"{}\".",
+                   mappingType, _accessor->meshContext(context.fromMeshID).mesh->getName(), _accessor->meshContext(context.toMeshID).mesh->getName());
 
       context.mapping->computeMapping();
     }
   }
+}
 
-  // Map data
-  for (impl::DataContext &context : _accessor->writeDataContexts()) {
+void SolverInterfaceImpl::mapData(utils::ptr_vector<DataContext> contexts, std::string mappingType)
+{
+  using namespace mapping;
+  MappingConfiguration::Timing timing;
+  for (impl::DataContext &context : contexts) {
     timing          = context.mappingContext.timing;
     bool hasMapping = context.mappingContext.mapping.get() != nullptr;
-    bool rightTime  = timing == MappingConfiguration::ON_ADVANCE;
-    rightTime |= timing == MappingConfiguration::INITIAL;
-    bool hasMapped = context.mappingContext.hasMappedData;
-    if (hasMapping && rightTime && (not hasMapped)) {
+    bool hasMapped  = context.mappingContext.hasMappedData;
+    bool mapNow     = timing == mapping::MappingConfiguration::ON_ADVANCE;
+    mapNow |= timing == mapping::MappingConfiguration::INITIAL;
```

Comment:
same as above regarding the namespace

Reply:
Fixed in d275cf93ebe63626e45d7394e975e34690cbec46.